samza-core/src/main/java/org/apache/samza/table/TableManager.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.table;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.JavaTableConfig;
+import org.apache.samza.serializers.KVSerde;
+import org.apache.samza.serializers.Serde;
+import org.apache.samza.storage.StorageEngine;
+import org.apache.samza.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link TableManager} manages all tables of a Samza job. For each table, it maintains
+ * the {@link TableSpec} and the {@link TableProvider}. It is used at execution for
+ * {@link org.apache.samza.container.TaskInstance} to retrieve table instances for
+ * read/write operations.
+ *
+ * A {@link TableManager} is constructed from job configuration, the {@link TableSpec}
+ * and {@link TableProvider} are constructed by processing the job configuration.
+ *
+ * After a {@link TableManager} is constructed, local tables are associated with
+ * local store instances created during {@link org.apache.samza.container.SamzaContainer}
```

Comment:
Each instance of {@link TableManager} is associated w/ local store instances within a TaskContext, right? Would be better to make it clear.

Reply:
Not really, local stores are associated with table instances. Table manager only passes store instances to table provider during init, after that it merely manages table instances. It is not aware of store instances direclty.